src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -1256,11 +1265,58 @@ void BluezObjectsCleanup(BluezEndpoint * apEndpoint)
     EndpointCleanup(apEndpoint);
 }
 
+#if !CHIP_BYPASS_ADDITIONAL_DATA_ADVERTISING
+static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * characteristic)
+{
+    if (characteristic == nullptr)
+    {
+        return;
+    }
+
+    // Construct the TLV for the additional data
+    GVariant * cValue = nullptr;
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+    TLVWriter writer;
+    TLVWriter innerWriter;
+    chip::System::PacketBufferHandle bufferHandle = chip::System::PacketBuffer::New();
+    chip::System::PacketBuffer * buffer           = bufferHandle.Get_ForNow();
+
+    writer.Init(buffer);
+
+    err = writer.OpenContainer(ProfileTag(kProtocol_ServiceProvisioning, kTag_AdditionalDataExensionDescriptor), kTLVType_Structure,
+                               innerWriter);
+    SuccessOrExit(err);
+
+    // Adding the rotating device id to the TLV data
+    err = innerWriter.PutString(ProfileTag(kProtocol_ServiceProvisioning, kRotatingDeviceIdTag), CHIP_ROTATING_DEVICE_ID);
```

Comment:
Does not match the spec.  Spec says to use `ContextTag(0)`, whereas this is using a profile-specific tag with tag value 1.

Reply:
updated